protein_lm/modeling/getters/training_args.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Dict, Union
+
+from pydantic import BaseModel, FieldValidationInfo, field_validator
+from transformers import TrainingArguments
+
+
+class TrainingArgsConfig(BaseModel):
+    per_device_train_batch_size: int
+    learning_rate: float
+    weight_decay: float
+    num_train_epochs: int
+    max_steps: int
+    save_steps: int
+    output_dir: str
+    save_strategy: str
+    report_to: str
+
+    @field_validator(
+        "per_device_train_batch_size",
+        "num_train_epochs",
+        "max_steps",
+        "weight_decay",
+        "learning_rate",
+        "save_steps",
+    )
+    @classmethod
+    def check_gt_zero(cls, v: Union[int, float], info: FieldValidationInfo):
+        if v <= 0:
+            raise ValueError(f"trainer.{info.field_name} must be greater than 0")
+        return v
+
+
+def get_training_args(config_dict: Dict) -> TrainingArguments:
+    config = TrainingArgsConfig(**config_dict)
+
+    if not os.path.isdir(config.output_dir):
+        print(f"creating checkpoint directory at {config.output_dir}")
+        os.makedirs(config.output_dir)
+
+    training_args = TrainingArguments(
```

Comment:
Small nit, why can't this also be like the dataset arguments, etc? And also why the separation between TrainingArgsConfig and TrainingArguments (from Huggingface)? Seems like the API is consistent

Reply:
I think you're raising some good points, though I'm not 100% sure I know what you're suggesting. Mainly it wasn't a very thoroughly-planned decision. 

* That there are two hard-coded arguments is just an oversight that I'll fix.
* At some point while writing this code, I wanted the validation that existed in the previous code for checking whether some params are nonnegative to remain in this version, since the purpose was a refactor. But upon reconsideration, I don't think this is important.
* I actually thought it would be more consistent to have a pydantic model (`TrainingArgsConfig`) rather than to just directly use huggingface's `TrainingArguments`, which appears to be just a dataclass. For instance, if we later decide to make a `TrainingConfig` pydantic model whose fields are all of the pydantic models I've introduced in this PR. But this is also not that important and could be changed later if it comes to that.

So in summary, if what you're suggesting is that I just return `TrainingArguments(**config_dict)` (and the accompanying changes to code) and get rid of `TrainingArgsConfig` entirely, I'm happy to make that change. Let me know if you meant something else!

Reply:
You also make good points. Having looked at the HF code's internals in the past few months, I would not be surprised if we would have to diverge from their conventions. It would also make it simper later on if we decide to use the flash attention models, for example.

I'm convinced now, I think the pydantic model is good and that we should go head with it.

Thanks again for all your work!

Reply:
Sounds good to me! I really don't have a strong preference either way, but of course happy to keep it as is.